### PR TITLE
Prettified flag usage strings.

### LINKED ISSUE #936: Replace verbose flag usage strings with help topics
We're using the flag usage strings wrong. Those strings are suppose to be short, 1-line descriptions of the flag. We have stuff like:

```
      --stores="": specify a comma-separated list of stores, specified by a colon-separated list of device attributes followed by '=' and either a filepath for a persistent store or an integer size in bytes for an in-memory store. Device attributes typically include whether the store is flash (ssd), spinny disk (hdd), fusion-io (fio), in-memory (mem); device attributes might also include speeds and other specs (7200rpm, 200kiops, etc.). For example, --stores=hdd:7200rpm=/mnt/hda1,ssd=/mnt/ssd01,ssd=/mnt/ssd02,mem=1073741824.
```

It would be better to add an `initflags` help topic with descriptions of the flags.

### PR DESCRIPTION
**Pull request by petermattis**
Ensured that flags were only added to commands that need them.

Fixes #936.

### CODE REVIEW

File: server/cli/flags.go [hunk 1]
```diff
@@ -57,71 +132,52 @@ func initFlags(ctx *server.Context) {
 		if strings.HasPrefix(f.Name, "test.") {
 			return
 		}
-		pf.Var(pflagValue{f.Value}, f.Name, f.Usage)
+		pf.Var(pflagValue{f.Value}, normalizeStdFlagName(f.Name), f.Usage)
```
Comment (tschottdorf) [inline server/cli/flags.go]: are there flags outside our control that have underscores?

File: server/cli/start.go [hunk 2]
```diff
@@ -77,35 +75,30 @@ func runInit(cmd *cobra.Command, args []string) {
...
-For example:
+Start a Cockroach node by joining the gossip network and exporting key ranges
+stored on physical device(s). The gossip network is joined by contacting one or
+more well-known hosts specified by the --gossip flag. Every node should be run
+with the same list of bootstrap hosts to guarantee a connected network. An
+alternate approach is to use a single host for --gossip and round-robin DNS.
 
-  cockroach start --gossip=host1:port1,host2:port2 --stores=ssd=/mnt/ssd1,ssd=/mnt/ssd2
+Each node exports data from one or more physical devices. These devices are
+specified via the --stores flag. This is a comma-separated list of paths to
```
Comment (tschottdorf) [inline server/cli/start.go]: This sounds like the format is `--stores=/tmp/a,/tmp/b`. We should mention the attributes.

**Comment by tschottdorf** [comment]
LGTM

File: server/cli/flags.go (hunk 1 above)
Comment (petermattis) [inline server/cli/flags.go]: The glog flags (e.g. `--log_dir`, `--log_backtrace_at`).

File: server/cli/start.go (hunk 2 above)
Comment (petermattis) [inline server/cli/start.go]: The attributes are mentioned in the `--stores` usage. I'll add a note to forward the reader to that description for more details.

**Pull request merged**